graviti/utility/lazy2.py

```diff
@@ -28,8 +28,35 @@ def __init__(self, total_count: int, limit: int) -> None:
         self.total_count = total_count
         self._limit = limit
 
-    def _init_offsets(self) -> None:
-        self._offsets = list(range(0, self.total_count, self._limit))
+    def _get_offsets(self) -> List[int]:
+        if not hasattr(self, "_offsets"):
+            self._offsets = list(range(0, self.total_count, self._limit))
+
+        return self._offsets
+
+    def setitem(self, start: int, stop: int, lengths: Iterable[int]) -> None:
+        """Update the offsets when setting items to the lazy list.
+
+        Arguments:
+            start: The start index.
+            stop: The stop index.
+            lengths: The length of the set values.
+
+        """
+        offsets = self._get_offsets()
+        partial_offsets = list(accumulate(lengths, initial=offsets[start]))
+        try:
+            last_offset = offsets[stop + 1]
+        except IndexError:
+            last_offset = self.total_count
+
+        diff = partial_offsets.pop() - last_offset
+        if diff != 0:
+            self.total_count += diff
+            for i in range(start + 1, len(offsets)):
+                offsets[i] += diff
+
+        offsets[start : stop + 1] = partial_offsets
```

Comment:
Why not update the offsets after setting the partial_offsets?

Reply:
because if the diff is not 0, the offsets needs to be updated first.